core/src/test/java/org/apache/iceberg/expressions/TestExpressionParser.java

```diff
@@ -0,0 +1,265 @@
+            "      \"value\" : \"2\\u0000\\u0000\\u0000\"\n" +
+            "    } ]\n" +
+            "  }\n" +
+            "}";
+
+    UnboundPredicate ltPredicate = new UnboundPredicate(
+            Expression.Operation.LT,
+            new NamedReference("Column1-Name"),
+            Literal.of(50));
+
+    Not notExpression = new Not(ltPredicate);
+    String actual = ExpressionParser.toJson(notExpression, true);
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testNestedExpression() {
+    String expected = "{\n" +
+            "  \"type\" : \"or\",\n" +
+            "  \"left-operand\" : {\n" +
+            "    \"type\" : \"and\",\n" +
+            "    \"left-operand\" : {\n" +
+            "      \"type\" : \"unbounded-predicate\",\n" +
+            "      \"operation\" : \"in\",\n" +
+            "      \"term\" : {\n" +
+            "        \"type\" : \"named-reference\",\n" +
+            "        \"value\" : \"Column1-Name\"\n" +
+            "      },\n" +
+            "      \"literals\" : [ {\n" +
+            "        \"type\" : \"integer\",\n" +
+            "        \"value\" : \"2\\u0000\\u0000\\u0000\"\n" +
+            "      } ]\n" +
+            "    },\n" +
+            "    \"right-operand\" : {\n" +
+            "      \"type\" : \"unbounded-predicate\",\n" +
+            "      \"operation\" : \"eq\",\n" +
+            "      \"term\" : {\n" +
+            "        \"type\" : \"named-reference\",\n" +
+            "        \"value\" : \"Column2-Name\"\n" +
+            "      },\n" +
+            "      \"literals\" : [ {\n" +
+            "        \"type\" : \"string\",\n" +
+            "        \"value\" : \"Test\"\n" +
+            "      } ]\n" +
+            "    }\n" +
+            "  },\n" +
+            "  \"right-operand\" : {\n" +
+            "    \"type\" : \"unbounded-predicate\",\n" +
+            "    \"operation\" : \"is_nan\",\n" +
+            "    \"term\" : {\n" +
+            "      \"type\" : \"named-reference\",\n" +
+            "      \"value\" : \"Column3-Name\"\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+
+    UnboundPredicate inPredicate = new UnboundPredicate(
+            Expression.Operation.IN,
+            new NamedReference("Column1-Name"),
+            Literal.of(50));
+
+    UnboundPredicate eqPredicate = new UnboundPredicate(
+            Expression.Operation.EQ,
+            new NamedReference("Column2-Name"),
+            Literal.of("Test"));
+
+    UnboundPredicate isNanPredicate = new UnboundPredicate(
+            Expression.Operation.IS_NAN,
+            new NamedReference("Column3-Name"));
+
+    And andExpression = new And(inPredicate, eqPredicate);
+    Or orNestedExpression = new Or(andExpression, isNanPredicate);
+    String actual = ExpressionParser.toJson(orNestedExpression, true);
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void aboveMaxAndBelowMinTest() {
```

Comment:
Style: test methods start with `test` and state what is being tested, rather than putting test at the end.

We also avoid putting multiple cases in the same test method because it serializes test results. If a test case fails and multiple cases are in the same method, then some tests aren't run. That's frustrating because when you fix a failing test and another case fails, you lost time running the suite instead of fixing multiple things. It is also possible that your fix broke another case in the method but you didn't know because you couldn't run both test cases.